src/ert/parsing/__init__.py

```diff
@@ -75,3 +77,21 @@ def replace(self, old: str, new: str, count=-1):
 ]
 
 Defines = List[List[str]]
+
+
+class ConfigWarning(UserWarning):
+    pass
+
+
+class ConfigValidationError(ValueError):
+    def __init__(self, errors: str, config_file: Optional[str] = None) -> None:
+        self.config_file = config_file
+        self.errors = errors
+        super().__init__(
+            (
+                f"Parsing config file `{self.config_file}` "
+                f"resulted in the errors: {self.errors}"
+            )
+            if self.config_file
+            else f"{self.errors}"
+        )
```

Comment:
`__init__.py` files should have an `__all__` field following https://peps.python.org/pep-0008/#public-and-internal-interfaces.